internal/transport/http2_server.go

```diff
@@ -306,15 +306,13 @@ func (t *http2Server) operateHeaders(frame *http2.MetaHeadersFrame, handle func(
 	state := &decodeState{
 		serverSide: true,
 	}
-	if err := state.decodeHeader(frame); err != nil {
-		if se, ok := status.FromError(err); ok {
-			t.controlBuf.put(&cleanupStream{
-				streamID: streamID,
-				rst:      true,
-				rstCode:  statusCodeConvTab[se.Code()],
-				onWrite:  func() {},
-			})
-		}
+	if h2code, err := state.decodeHeader(frame); err != nil {
```

Comment:
The earlier code seems to be adding a `cleanupStream` to the `controlBuf` only if the returned error was a `status.Error`. That requirement seems to have been lifted with the new code. Was this intentional? If so, why?

Reply:
This wasn't intentional, thanks for the catch, revert it back to original behavior.